### PR TITLE
Add outbound

### PR DESCRIPTION
**Pull request by ascasson**
Includes Outbound Link module. Makes gtag available via the window object.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+        if (props.target && props.target.toLowerCase() !== `_self`) {
+          redirect = false;
+        }
+
+        if (window.gtag) {
+          window.gtag('event', 'click', {
+            event_category: 'outbound',
+            event_label: props.href,
+            transport_type: 'beacon',
```
Comment (flpvsk) [inline src/index.js:36]: In the original `OutboundLink` transport depends on whether it is a redirect or not. Should that also be the case here? https://github.com/gatsbyjs/gatsby/blob/45ec012f03a533c28b4ebf3b944f2342a7bd86f2/packages/gatsby-plugin-google-analytics/src/index.js#L31

**Review by flpvsk — COMMENTED** [review]
I've noticed one difference with the original OutboundLink (see comment). Is it intended?

File: src/index.js (hunk 1 above)
Comment (ascasson) [reply src/index.js:36]: @flpvsk That's what's generally shared for anyone tracking outbound links. Based on this reference to the old analytics.js api, I suspect having beacon set is the norm: "in the future, analytics.js will likely switch to using 'beacon' as the default mechanism in browsers that support it." Overall it's kind of vague. Thoughts?

Comment (flpvsk) [reply src/index.js:36]: @ascasson alright, I agree. Makes sense to keep it as the default.

**Pull request merged**